### ISSUE TITLE
[no-unsafe-assignment/return] Relax restrictions to allow any => unknown assignment

### ISSUE THREAD

**Issue by clar-cmp** [typescript-eslint/typescript-eslint#2325]
**Repro**

```TS
function myFunction(): unknown {
    // triggers no-unsafe-assignment because JSON.parse returns any
    let val: unknown = JSON.parse(someStringVar);

    // triggers no-unsafe-return because JSON.parse returns any
    return JSON.parse(someStringVar);
}
```

**Expected Result**

This should not throw an error because a direct any => unknown cast cannot possibly be unsafe. Requiring an `as unknown` everywhere is unnecessary noise.

**Additional notes**

Ideally, `any` should not be used across a code base, and I like that the current default rules enforce this. However, a lot of existing APIs still use `any`, including the TypeScript library itself, and we have to make do with what we have.

**Comment by bradzacher** [comment]
Happy to accept a PR!

**Comment by yasarsid** [comment]
Hi @bradzacher - If its all right, I want to work on this.

**Comment by maletor** [comment]
This is awesome!

But I got down a rabbit hole where you can't take 'unknown' back to a structured object like an interface. 

You can take 'unknown' to 'object' but you're stuck there. Even doing 'prop in object' doesn't work.

Whats the reason? Is there a workaround?

**Comment by maletor** [comment]
Addendum: I found user defined type guards but those seem really error prone. What happens if the interface changes and the type guard doesn't? Seems like it's of little use.

**Comment by bradzacher** [comment]
In this regard, unknown and any are the same. The difference being that unknown ensures you don't do things that are unsafe.

Not being able to convert past object is a known issue that they haven't solved just yet. https://github.com/microsoft/TypeScript/issues/21732

Essentially though, yes, you've run into the issue with typing an untyped boundary. There is no simple way to do this unfortunately. There are some libraries to help generate validation functions from type definitions, but there's no perfect solution.

**Comment by maletor** [comment]
I'll give 'runtypes' a try but it seems like #21732 may be close after a hard long two years.
